cli/internal/cmd/apply.go

```diff
@@ -545,6 +546,16 @@ func (a *applyCmd) validateInputs(cmd *cobra.Command, configFetcher attestationc
 					return nil, nil, fmt.Errorf("aborted by user")
 				}
 			}
+
+			validVersionString, err := versions.ResolveK8sPatchVersion(xsemver.MajorMinor(string(conf.KubernetesVersion)))
+			if err != nil {
+				return nil, nil, fmt.Errorf("resolving Kubernetes patch version: %w", err)
+			}
+			validVersion, err = versions.NewValidK8sVersion(validVersionString, true)
+			if err != nil {
+				return nil, nil, fmt.Errorf("parsing Kubernetes version: %w", err)
+			}
+
```

Comment:
This needs to be moved outside of the skipK8s if block. Otherwise it does not solve the issue. This is one of the reasons I wanted to rename the `contains` method to `skip`;)

Reply:
https://github.com/edgelesssys/constellation/commit/3ef8cee8612b86ef5b1ef64901d43b749393b76a

Reply:
I thought the contents of the KubernetesVersion field in the config are irrelevant otherwise, since the upgrade phase is skipped. However, thinking about it again and looking at the code, it seems to me that we should, as Adrian mentioned, adjust it in both cases. The config needs to contain a correct version once the service versions are selected

Reply:
yes, this is relevant for helm since it uses the K8s version to determine the chart config.